engine/language-server/src/main/scala/org/enso/languageserver/boot/ProfilingConfig.scala

```diff
@@ -23,19 +23,68 @@ case class ProfilingConfig(
     * @return the path to the runtime events log file
     */
   def profilingEventsLogPath: Option[Path] =
-    profilingPath.map { path =>
-      val profilingDirectory     = path.getParent
-      val profilingFileName      = path.getFileName.toString
-      val profilingFileExtension = FilenameUtils.getExtension(profilingFileName)
-      val eventsLogFileName =
-        profilingFileName.stripSuffix(
-          profilingFileExtension
-        ) + ProfilingConfig.EventsLogExtension
-
-      profilingDirectory.resolve(eventsLogFileName)
-    }
+    profilingPath.map(
+      ProfilingConfig.modifyPath(
+        _,
+        ProfilingConfig.EventsLogSuffix,
+        ProfilingConfig.EventsLogExtension
+      )
+    )
+
+  /** Creates a path to the JSON-RPC text messages log. */
+  def textMessagesPath: Option[Path] =
+    profilingPath.map(
+      ProfilingConfig.modifyPath(
+        _,
+        ProfilingConfig.MessagesSuffix,
+        ProfilingConfig.TextMessagesExtension
+      )
+    )
+
+  /** Creates a path to the binary messages log. */
+  def binaryMessagesPath: Option[Path] =
+    profilingPath.map(
+      ProfilingConfig.modifyPath(
+        _,
+        ProfilingConfig.MessagesSuffix,
+        ProfilingConfig.BinaryMessagesExtension
+      )
+    )
 }
+
 object ProfilingConfig {
 
+  private val EventsLogSuffix    = ""
   private val EventsLogExtension = "log"
+
+  private val MessagesSuffix          = "-messages"
+  private val TextMessagesExtension   = "txt"
+  private val BinaryMessagesExtension = "bin"
```

Comment:
Does it make sense to split textual and binary messages? How are we going to replay them and keep order?